lib/main.js

```diff
@@ -39,6 +39,11 @@ function parse (src /*: string | Buffer */, options /*: ?DotenvParseOptions */)
 
   // convert Buffers before splitting into lines and processing
   src.toString().split(NEWLINE).forEach(function (line, idx) {
+    line = line.trim()
+    // ignore empty and commented lines
```

Comment:
if you want to influence "debug" behavior, I'd suggest moving closer there instead of possibly altering the parsing logic. `if (debug && line.length && line[0] !== '#')` might do it, although I have reservations about _decreasing_ debug output.